v4/feature/workers/src/main/java/exchange/dydx/trading/feature/workers/globalworkers/DydxGasTokenWorker.kt

```diff
@@ -0,0 +1,39 @@
+package exchange.dydx.trading.feature.workers.globalworkers
+
+import exchange.dydx.abacus.state.manager.GasToken
+import exchange.dydx.dydxstatemanager.AbacusStateManagerProtocol
+import exchange.dydx.trading.feature.shared.PreferenceKeys
+import exchange.dydx.utilities.utils.Logging
+import exchange.dydx.utilities.utils.SharedPreferencesStore
+import exchange.dydx.utilities.utils.WorkerProtocol
+import kotlinx.coroutines.CoroutineScope
+
+private const val TAG = "DydxGasTokenWorker"
+
+class DydxGasTokenWorker(
```

Comment:
why not inject this?

Reply:
We can refactor this later... it involves changes to the parent worker as well.